### PR TITLE
Add better android platform cache busters

### PR DESCRIPTION
**Pull request by vmrob**
Instead of relying entirely on compiler arguments for android configuration hashes, this patch provides a dependence on the specific version of the api and runtime projects are built for which results in a much more accurate set of cached objects and cache busting when those versions change.

### CODE REVIEW

**Comment by vmrob** [comment]
I'm much happier with this than I was with the state of Android caching before. Here's an example before/after:

```
    webglnoise      out-of-date
    --- before
    +++ after
    @@ -15,6 +15,7 @@
    @@ -22,6 +23,11 @@
             "repository": "https://github.com/ashima/webgl-noise.git"
         },
         "platform-configuration": {
    +        "api-level": 17,
             "c-compiler": [
                 "arm-linux-androideabi-clang",
                 "-march=armv7-a"
             "cxx-compiler": [
                 "arm-linux-androideabi-clang++",
                 "-march=armv7-a"
    -        ]
    +        ],
    +        "runtime": "libstdc++",
    +        "runtime-version": {
    +            "libc++-abi-version": "1",
    +            "libc++-version": "1101"
    +        }
         }
     }
```

For reference, the abi version for both NDKs we use is the same, but the version number goes from 1101 to 3900--wow.

File: needy/platforms/android.py [hunk 1]
```diff
@@ -145,3 +148,73 @@ def ndk_home(self):
...
+        compiler = self.cxx_compiler(architecture)
+        binary_paths = self.binary_paths(architecture)
+        d = Platform.configuration(self, architecture)
+        d.update({
+            'api-level': self.__android_api_level(compiler, binary_paths),
+            'runtime': self.__runtime,
+            'runtime-version': {
+                'libc++-version': self.__libcpp_version(compiler, binary_paths),
+                'libc++-abi-version': self.__libcpp_abi_version(compiler, binary_paths)
+            } if self.__runtime == 'libstdc++' else {}
```
Comment (ccbrown) [inline needy/platforms/android.py]: and down here, the runtimes that use libc++ would be "c++_static" or "c++_shared". but in the case of standalone toolchains, you don't necessarily need to specify the runtime argument (so this should still work when `self.__runtime is None`

so i'd like to just make the case where `_LIBCPP_VERSION` is absent an expected case, with no warnings and good behavior

**Comment by ccbrown** [comment]
For reference: https://developer.android.com/ndk/guides/cpp-support.html

Sorry, but you're touching kinda sketchy code to begin with, so I'd like to clean it up and not make it worse if it's not too much to ask.

**Comment by vmrob** [comment]
Yeah, I was aware that there are multiple libc++ runtimes; I was just sticking to what I thought were the supported variants. I can go through and add each of those, but I think, instead of supporting gnustl, I would rather it error out and say that isn't currently supported--especially considering that is going away in the future of the ndk.

Sound reasonable?

**Pull request merged**